### PR TITLE
 advance search form

### PR DESCRIPTION
**Pull request by samnangcattor**
- advance search 
- search with address
- search with price
- search with square
- search wtih keyword
- search with facility
- search with parking

### CODE REVIEW

File: app/views/addresses/_address.html.erb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  <div class="address-details">
+    <div class="title">
+      <%= link_to address.title, address_path(address),
+      title: "#{address.title}" %>
+      <span class="right"><%= address.created_at.to_date.to_formatted_s(:long_ordinal) %>
+      </span>
+    </div>
+    <p><%= address.address %></p>
+    <p><%= t("capacity") %>: <span class="capacity"><%= address.capacity %>
+    </span> | <%= t("contact") %>: <span class="contact"><%= address.contact %>
```
Comment (HieuMeo) [inline app/views/addresses/_address.html.erb]: Remove " "

File: app/views/searchs/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<h3><%= t"search.result" %></h3>
```
Comment (HieuMeo) [inline app/views/searchs/index.html.erb]: Remove " "

File: app/assets/javascripts/application.js [hunk 3]
```diff
@@ -24,10 +24,20 @@ $(document).ready(function(){
     case "new_address":
       getElementById("new_address_view").style.display="block";
   }
+
+  $(".chb1").change(function(){
```
Comment (dieunb) [inline app/assets/javascripts/application.js]: gộp 2 hàm này vào 1

File: app/views/searchs/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+<div class="panel panel-info">
+  <div class="panel-heading">
+    <%= render @addresses %>
+    <% if @addresses != [] %>
```
Comment (dieunb) [inline app/views/searchs/index.html.erb]: `@addresses.blank?`

File: app/views/searchs/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<h3><%= t"search.result" %></h3>
+<div class="panel panel-info">
+  <div class="panel-heading">
+    <%= render @addresses %>
```
Comment (dieunb) [inline app/views/searchs/index.html.erb]: move to `if` condition

File: app/assets/javascripts/application.js [hunk 6]
```diff
@@ -24,10 +24,83 @@ $(document).ready(function(){
...
+      if(min_square === "" && max_square === "" &&
+        min_facility == "" && max_facility === "" &&
+        min_price === "" && max_price === "" && 
+        address === "" && title === "" && !$(".checkbox").is(":checked")){
+          event.preventDefault(); 
+      }
+  });
+
+  
+  $("select").change(function(){
```
Comment (cuongnguyen2503) [inline app/assets/javascripts/application.js]: Can use switch case instead if else.

File: app/assets/javascripts/application.js [hunk 7]
```diff
@@ -24,10 +24,83 @@ $(document).ready(function(){
...
+    var value = $("#sorting option:selected").val();
+    alert(value);
+    $.get("searchs", {});
+    
+  });
 });
 
+function min_change(name){
+  var text_select = parseInt($("#min_"+ name +" option:selected").text(), 10);
+  var max_lenght = $("#max_"+ name +" option").size();
```
Comment (cuongnguyen2503) [inline app/assets/javascripts/application.js]: Space before "+" here and somewhere.

File: app/assets/javascripts/application.js (hunk 6 above)
Comment (samnangcattor) [inline app/assets/javascripts/application.js]: Everything is fixed  but your comment, it is not right place

File: app/assets/javascripts/application.js [hunk 8]
```diff
@@ -25,10 +25,83 @@ $(document).ready(function(){
...
 
+function min_change(name){
+  var text_select = parseInt($("#min_" + name + " option:selected").text(), 10);
+  var max_lenght = $("#max_" + name + " option").size();
+
+  $("#max_"+ name)[0].selectedIndex = 0;
+
+  for(var i = 0; i < max_lenght; i++){
+    var text_collect = parseInt($("#max_" + name).find("option").eq(i).text(), 10);
+
```
Comment (tranhuyhoangbka) [inline app/assets/javascripts/application.js]: space

File: app/assets/stylesheets/custom.css.scss [hunk 9]
```diff
@@ -524,3 +527,36 @@ legend {
...
   margin-left: 5px;
 }
+
+select.form-control{
+  -webkit-appearance: none;
+  -moz-appearance: none;
+  appearance: none;
+  height: 25px;
+  padding: 2px;
+  background: url(arrow.png) no-repeat right;
```
Comment (tranhuyhoangbka) [inline app/assets/stylesheets/custom.css.scss]: i think it is url(assets/arrow.png)
or image-url("arrow.png")

File: app/assets/stylesheets/custom.css.scss [hunk 10]
```diff
@@ -524,3 +527,36 @@ legend {
 #fa-usd {
   margin-left: 5px;
 }
+
+select.form-control{
```
Comment (tranhuyhoangbka) [inline app/assets/stylesheets/custom.css.scss]: you should make one space beetween selector-css and "{"

File: app/views/rooms/_room.html.erb [hunk 11]
```diff
@@ -0,0 +1,46 @@
+<div class="address">
+  <div class="address-thumb">
+    <% if room.address.images.first.present? %>
+      <%= image_tag room.address.images.first.photo.small_thumb.url %>
+    <% else %>
+      <%= image_tag "house-thumb.jpg" %>
```
Comment (tranhuyhoangbka) [inline app/views/rooms/_room.html.erb]: "house-thumb.jpg" get it on to settings global

File: app/views/rooms/_room.html.erb [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    <% end %>
+    <div class="address-id">
+      <%= t "search.range" %>
+      <%= room.address.id %>
+    </div>
+  </div>
+  <div class="address-details">
+    <div class="title">
+      <%= link_to room.address.title, address_path(room.address),
+        title: "#{room.address.title}" %>
```
Comment (tranhuyhoangbka) [inline app/views/rooms/_room.html.erb]: only need     title: room.address.title

File: app/views/rooms/_room.html.erb (hunk 11 above)
Comment (samnangcattor) [inline app/views/rooms/_room.html.erb]: cai nay khong phan la phan  cua minh het  , neu  minh chinh sua  local of  this file  , it is effact to other  , example   app/view/rooms/_address.html.erb

File: app/views/rooms/_room.html.erb (hunk 12 above)
Comment (samnangcattor) [inline app/views/rooms/_room.html.erb]: Are you sure ?

File: app/views/rooms/_room.html.erb [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+    <p>
+      <%= t "capacity" %>: <span class="capacity"><%= room.address.capacity %>
+      </span> |
+      <%= t "contact" %>: <span class="contact"><%= room.address.contact %>
+      </span> | <%= t "province_city" %>:
+        <span class="province_city"><%= room.address.region.province %></span>
+    </p>
+    <div>
+      <span class="right">
+        <%= room.address.created_at.to_date.to_formatted_s(:long_ordinal) %>
```
Comment (tranhuyhoangbka) [inline app/views/rooms/_room.html.erb]: remove ()

File: app/views/shared/_condition_field.html.erb [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+          :last, :last, {include_blank: t("search.maxmum")},
+          {class: "form-control", id: "max_facility"} %>
+      </div>
+
+      <div class="col-sm-3">
+        <%= f.label :address_region_province %>
+        <%= f.collection_select :address_region_province_cont, Settings.locations,
+          :last, :last, {include_blank: t("search.region")},
+          {class: "form-control", id: "address"} %>
+
```
Comment (tranhuyhoangbka) [inline app/views/shared/_condition_field.html.erb]: remove space

File: app/views/shared/_condition_field.html.erb [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+        <div class="col-xs-6">
+          </p><%= f.check_box :parking_in, {class: "checkbox", id: "park2"}, 0, nil %>
+          <div id="none"><%= f.label :none %></div>
+        </div>
+      </div>
+    </div>
+
+    <div class="row-mid">
+      </p><%= f.text_field :address_title_cont, {placeholder: t("search.title"),
```
Comment (tranhuyhoangbka) [inline app/views/shared/_condition_field.html.erb]: remove </p> at header line

File: app/views/shared/_condition_field.html.erb [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+        class: "form-control", id: "title"} %><p></p>
+      <div class="ad_search">
+        <%= f.submit t("api.search"), {class: "btn btn-info btn-block", id: "sumbit_search"} %>
```
Comment (tranhuyhoangbka) [inline app/views/shared/_condition_field.html.erb]: remove {} at last hash of method

**Pull request merged**